ee/server/service/software_installers.go

```diff
@@ -429,6 +444,18 @@ func (svc *Service) UpdateSoftwareInstaller(ctx context.Context, payload *fleet.
 		payload.UninstallScript = &uninstallScript
 	}
 
+	fieldsShouldSideEffect := map[string]bool{
```

Comment:
Seems like this should be a map of empty structs so we don't have unset vs. false vs. true?

Reply:
Agreed, that makes more sense and makes the looping logic a lil cleaner as well.